### PR TITLE
Multiple test methods do not work when extending CamelQuarkusTestSupport

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4560
fixes https://github.com/apache/camel-quarkus/issues/4569

Brief summary of changes:

Problem was caused by the difference between lifecycle of camel and camel-quarkus. As the camel context is created once, there are several callbacks, which executes context "resetting" so it behave likes it starts for each test again. During the resetting, all routes were stopped and removed. Unfortunately in a scenario with global routeBuilder (like the reproducer from the issue) it meant that  even routes from tat global routeBuilder were removed ,  and the next test classes (or methods) missed routes from that routeBuilder. This PR changes the behavior in a way, that only routes added during setup of the test are removed. Therefore there is a local variable `createdRoutes` which , holds routes created during the test setup.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CallbackUtil.java [hunk 1]
```diff
@@ -46,15 +46,22 @@ static Optional<TestInstance.Lifecycle> getLifecycle(CamelQuarkusTestSupport tes
...
+        //if routeBuilder (from the test) was used, all routes from that builder has to be stopped and removed
         //because routes will be created again (in case of TestInstance.Lifecycle.PER_CLASS, this method is not executed)
-        if (testInstance.isUseRouteBuilder()) {
-            try {
-                testInstance.context().getRouteController().stopAllRoutes();
-                testInstance.context().getRouteController().removeAllRoutes();
+        if (testInstance.isUseRouteBuilder() && testInstance.createdRoutes != null) {
 
+            try {
+                if (keepOriginalRoutes) {
```
Comment (aldettinger) [inline test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CallbackUtil.java:56]: Could you please details what `keepOriginal` means in this context ?

It's like we keep the state of route ? Or we want a fresh route ?

Comment (JiriOndrusek) [reply test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CallbackUtil.java:56]: I'll add comment here. that makes sense, it didn't occur to me that it is my own definition of routes.

Comment (JiriOndrusek) [reply test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CallbackUtil.java]: I removed this if. It should not be there. There is a following ticket, which I'm starting to work on, which takes care of the lifecycle with Lifecycle.PER_CLASS - https://github.com/apache/camel-quarkus/issues/4569

**Review by aldettinger — APPROVED** [review]
Ok, many thanks for adjustement :+1:

**Pull request merged**